src/index.js

```diff
@@ -20,143 +29,16 @@ const valuesToConvert = arrayToObject([
-        // intentionally only grabbing the first instance of this because that represents `left`
-        .replace(/^((-|\d|\.)+%)/, (match, group) =>
-          calculateNewBackgroundPosition(group),
-        )
-        .replace(/(left)|(right)/, match => valuesToConvert[match])
-    )
-  },
-  backgroundPositionX(value) {
-    if (isNumber(value)) {
-      return value
-    }
-    return propertyValueConverters.backgroundPosition(value)
-  },
-  transform(value) {
-    // This was copied and modified from CSSJanus:
-    // https://github.com/cssjanus/cssjanus/blob/4a40f001b1ba35567112d8b8e1d9d95eda4234c3/src/cssjanus.js#L152-L153
-    const nonAsciiPattern = '[^\\u0020-\\u007e]'
-    const unicodePattern = '(?:(?:\\[0-9a-f]{1,6})(?:\\r\\n|\\s)?)'
-    const numPattern = '(?:[0-9]*\\.[0-9]+|[0-9]+)'
-    const unitPattern = '(?:em|ex|px|cm|mm|in|pt|pc|deg|rad|grad|ms|s|hz|khz|%)'
-    const escapePattern = `(?:${unicodePattern}|\\\\[^\\r\\n\\f0-9a-f])`
-    const nmstartPattern = `(?:[_a-z]|${nonAsciiPattern}|${escapePattern})`
-    const nmcharPattern = `(?:[_a-z0-9-]|${nonAsciiPattern}|${escapePattern})`
-    const identPattern = `-?${nmstartPattern}${nmcharPattern}*`
-    const quantPattern = `${numPattern}(?:\\s*${unitPattern}|${identPattern})?`
-    const signedQuantPattern = `((?:-?${quantPattern})|(?:inherit|auto))`
-    const translateXRegExp = new RegExp(
-      `(translateX\\s*\\(\\s*)${signedQuantPattern}(\\s*\\))`,
-      'gi',
-    )
-    const translateRegExp = new RegExp(
-      `(translate\\s*\\(\\s*)${signedQuantPattern}((?:\\s*,\\s*${signedQuantPattern}){0,1}\\s*\\))`,
-      'gi',
-    )
-    const translate3dRegExp = new RegExp(
-      `(translate3d\\s*\\(\\s*)${signedQuantPattern}((?:\\s*,\\s*${signedQuantPattern}){0,2}\\s*\\))`,
-      'gi',
-    )
-    return value
-      .replace(translateXRegExp, calculateNewTranslate)
-      .replace(translateRegExp, calculateNewTranslate)
-      .replace(translate3dRegExp, calculateNewTranslate)
-  },
-}
-propertyValueConverters.margin = propertyValueConverters.padding
-propertyValueConverters.borderWidth = propertyValueConverters.padding
-propertyValueConverters.boxShadow = propertyValueConverters.textShadow
-propertyValueConverters.webkitBoxShadow = propertyValueConverters.textShadow
-propertyValueConverters.mozBoxShadow = propertyValueConverters.textShadow
-propertyValueConverters.borderStyle = propertyValueConverters.borderColor
-propertyValueConverters.webkitTransform = propertyValueConverters.transform
-propertyValueConverters.mozTransform = propertyValueConverters.transform
+// Sorry for the regex 😞, but basically thisis used to replace _every_ instance of
+// `ltr`, `rtl`, `right`, and `left` in `backgroundimage` with the corresponding opposite.
+// A situation we're accepting here:
+// url('/left/right/rtl/ltr.png') will be changed to url('/right/left/ltr/rtl.png')
+// Definite trade-offs here, but I think it's a good call.
+const bgImgDirectionRegex = new RegExp(
```

Comment:
Could these RegExps (as well as those defined in `propertyValueConverters.transform`) be defined in a separate constants file? Then the individual converter methods can import them as needed, instead of passing them through on line 113 below.

Reply:
They need to be passed through, because different consumers will use different RegExps.
They can still be defined in a separate constants file.

The RegExp in`propertyValueConverters.transform` was left in place because I couldn't see how it may become implementation specific.